zebra-consensus/src/transaction.rs

```diff
@@ -161,6 +161,16 @@ where
 
             if tx.is_coinbase() {
                 check::coinbase_tx_no_prevout_joinsplit_spend(&tx)?;
+            } else {
+                // check the value pool for non-coinbase transactions
+                if tx
+                    .value_balance(&req.known_utxos())
```

Comment:
We need to implement #2440 before we add this check, so we have all the required UTXOs here.

Reply:
Actually, #2440 isn't enough to give us the correct v5 transaction UTXOs.

A malicious user can create a chain fork with the same transaction inputs (and v5 transaction ID), but different transparent outputs. (Or the same transparent outputs in a different order.)

We only have the correct information to do the check in the non-finalized chain:
https://github.com/ZcashFoundation/zebra/pull/2525/files#r677188739

Reply:
We should make this move after https://github.com/ZcashFoundation/zebra/pull/2525 is merged.

Reply:
This check needs to get all the UTXOs anyway, so we can't merge the PR with it here - Zebra will crash.

We could do this check in the non-finalized state after `transparent_double_spends` is checked:
https://github.com/ZcashFoundation/zebra/blob/236388909e1a66ecdc5a4aa0f5e982e4e2f36ba7/zebra-state/src/service/non_finalized_state.rs#L167-L180

Reply:
Blocking: This check needs to be moved or removed before we can merge, because it sometimes makes Zebra panic:
```
   The application panicked (crashed).
   Message:  Provided Utxos don't have transaction Outpoint
   Location: zebra-chain/src/transparent.rs:152
```
https://github.com/ZcashFoundation/zebra/pull/2486/checks?check_run_id=3175320450#step:11:584

The panic happens when Zebra tries to verify an gossiped block:
> zebrad::components::inbound::downloads::download_and_verify
https://github.com/ZcashFoundation/zebra/pull/2486/checks?check_run_id=3175320450#step:11:598